Objects/typeobject.c

```diff
@@ -3741,6 +3741,91 @@ type_is_gc(PyTypeObject *type)
     return type->tp_flags & Py_TPFLAGS_HEAPTYPE;
 }
 
+
+// TODO: MM: Move the following three functions?
+static int
+is_typing_name(PyObject *obj, char *name)
+{
+    PyTypeObject *type = Py_TYPE(obj);
+    if (strcmp(type->tp_name, name) != 0) {
+        return 0;
+    }
+    PyObject *module = PyObject_GetAttrString((PyObject *)type, "__module__");
+    if (module == NULL) {
+        return -1;
+    }
+    int res = PyUnicode_Check(module)
+        && _PyUnicode_EqualToASCIIString(module, "typing");
+    Py_DECREF(module);
+    return res;
+}
+
+static int
+is_typevar(PyObject *obj)
+{
+    return is_typing_name(obj, "TypeVar");
+}
+
+static int
+is_genericalias(PyObject *obj)
+{
+    return is_typing_name(obj, "GenericAlias");
+}
+
+static int
+is_new_type(PyObject *obj)
+{
+    if (!PyObject_IsInstance((PyObject *)obj, (PyObject *)&PyFunction_Type)) {
+        return 0;
+    }
+    PyObject *module = PyObject_GetAttrString((PyObject *)obj, "__module__");
+    if (module == NULL) {
+        return 0;
+    }
+    return PyUnicode_Check(module) && _PyUnicode_EqualToASCIIString(module, "typing");
+}
+
+static int
+is_not_unionable (PyObject *obj)
```

Comment:
```suggestion
is_not_unionable(PyObject *obj)
```
Also consider changing the sense -- having a function `is_unionable()` makes more sense. The callers then just add `!` in front.